src/main/java/com/redhat/foreman/ForemanSharedNodeCloud.java

```diff
@@ -90,6 +90,7 @@
 
     private transient ForemanAPI api = null;
     private transient ForemanComputerLauncherFactory launcherFactory = null;
+    private transient Map<String, String> hostsMap = new HashMap<String, String>();
```

Comment:
Isn't more save and robust use a `java.util.concurrent.atomic.AtomicReference<V>` here?